packages/attune/tests/smoke/main.rs

```diff
@@ -0,0 +1,465 @@
+#[derive(Debug, Clone)]
+struct TestConfig {
+    cli_path: String,
+    api_endpoint: String,
+    api_token: String,
+    gpg_key_id: String,
+}
+
+#[test]
+fn smoke_tests() {
+    println!("\n========== RUNNING ATTUNE CLI SMOKE TESTS ==========");
+    
+    // Set up configuration upfront
+    let sh = Shell::new().unwrap();
+    
+    // Get CLI binary path
+    let cli_path = env::var("ATTUNE_CLI_PATH").expect(
+        "ATTUNE_CLI_PATH environment variable is required. Set it to the path of your CLI binary.\n\
+         Example: export ATTUNE_CLI_PATH=/target/release/attune",
+    );
+    
+    // Get API configuration
+    let api_token = env::var("ATTUNE_API_TOKEN").unwrap_or_else(|_| "INSECURE_TEST_TOKEN".to_string());
+    let api_endpoint = env::var("ATTUNE_API_ENDPOINT").unwrap_or_else(|_| "http://localhost:3000".to_string());
+    
+    // Set up GPG key for testing
+    println!("\n========== SMOKE TEST: Setup GPG Key =========");
+    let gpg_key_id = setup_gpg_key(&sh);
+    
+    // Create test configuration
+    let config = TestConfig {
+        cli_path,
+        api_endpoint,
+        api_token,
+        gpg_key_id,
+    };
+    
+    println!("Test configuration:");
+    println!("  CLI Path: {}", config.cli_path);
+    println!("  API Endpoint: {}", config.api_endpoint);
+    println!("  API Token: {}", config.api_token);
+    println!("  GPG Key ID: {}", config.gpg_key_id);
+
+    // Run tests in order.
+    println!("\n========== SMOKE TEST: Check Attune CLI Exists =========");
+    test_cli_binary_exists(&config);
+    println!("\n========== SMOKE TEST: Repository Create (2 repos) =========");
+    test_repo_create_multiple(&config);
+    println!("\n========== SMOKE TEST: Repository List (verify both repos) =========");
+    test_repo_list_multiple(&config);
+    println!("\n========== SMOKE TEST: Repository Delete =========");
+    test_repo_delete(&config);
+    println!("\n========== SMOKE TEST: Verify Repository Deleted =========");
+    test_repo_delete_verification(&config);
+    println!("\n========== SMOKE TEST: Package Add =========");
+    test_pkg_add(&config);
+    println!("\n========== SMOKE TEST: Package Delete =========");
+    test_pkg_delete(&config);
+    
+    println!("\n========== ALL SMOKE TESTS COMPLETED SUCCESSFULLY ==========");
+}
+
+/// Helper function to set up shell environment variables from config.
+fn set_up_shell_env(sh: &Shell, config: &TestConfig) {
+    sh.set_var("ATTUNE_API_TOKEN", &config.api_token);
+    sh.set_var("ATTUNE_API_ENDPOINT", &config.api_endpoint);
+}
+
+/// Helper function to set up GPG key for testing.
+fn setup_gpg_key(sh: &Shell) -> String {
+    println!("Setting up GPG key for testing...");
+    
+    // Create GPG key configuration file
+    let gpg_config = r#"Key-Type: EDDSA
+Key-Curve: Ed25519
+Subkey-Type: ECDH
+Subkey-Curve: Cv25519
+Name-Real: Attune Test
+Name-Email: test@attunehq.com
+Expire-Date: 0
+%no-protection
+%commit
+"#;
```

Comment:
[optional] you can use https://docs.rs/indoc/latest/indoc/ to indent the content (so it looks like it fits in the function) without leading spaces.